runners/flink/src/main/java/org/apache/beam/runners/flink/translation/functions/FlinkExecutableStageFunction.java

```diff
@@ -230,20 +330,92 @@ public void close() throws Exception {
     private final Collector<RawUnionValue> collector;
 
     private final Map<String, Integer> outputMap;
+    @Nullable private final TimerReceiverFactory timerReceiverFactory;
 
     ReceiverFactory(Collector<RawUnionValue> collector, Map<String, Integer> outputMap) {
+      this(collector, outputMap, null);
+    }
+
+    ReceiverFactory(
+        Collector<RawUnionValue> collector,
+        Map<String, Integer> outputMap,
+        @Nullable TimerReceiverFactory timerReceiverFactory) {
       this.collector = collector;
       this.outputMap = outputMap;
+      this.timerReceiverFactory = timerReceiverFactory;
     }
 
     @Override
     public <OutputT> FnDataReceiver<OutputT> create(String collectionId) {
       Integer unionTag = outputMap.get(collectionId);
-      checkArgument(unionTag != null, "Unknown PCollection id: %s", collectionId);
-      int tagInt = unionTag;
+      if (unionTag != null) {
+        int tagInt = unionTag;
+        return receivedElement -> {
+          synchronized (collectorLock) {
+            collector.collect(new RawUnionValue(tagInt, receivedElement));
```

Comment:
Yes, but I don't know if it is applicable for batch since it is scheduled stage by stage. For streaming the issue would surface just by running wordcount, for batch not.

Reply:
The stage-by-stage is only in place if the `ExecutionMode` is set to `FORCE_BATCH`, instead of the default `PIPELINE`.